integration-tests/itest/deploy_test.go

```diff
@@ -0,0 +1,306 @@
+)
+
+// Assumptions:
+// In this test, we are going to deploy the test application to an existing cluster.
+// For this first step, we assume that the cluster is correctly deployed and that kubectl is configured to talk to it.
+// The application is already built
+// We also assume that the test is going to be executed manually.
+// Automation will be the next step
+
+const (
+	JsonTokenSrc    = "/keybase/team/assassins/gcloud/pipelines-serviceaccount-key-container-registry-read-write.json"
+	ShortTimeout    = 60  // seconds
+	LongTimeout     = 240 // seconds
+	InitialWaitTime = "30s"
+)
+
+var swissKnifeApp = cli.App{
+	Image: "eu.gcr.io/bubbly-observer-178213/swiss-knife:189-277e424",
+	Name:  "swiss-knife",
+}
+
+var _ = Describe("Application deployment", func() {
+	Context("the cluster is clean for testing", func() {
+		It("should not have the test app", func() {
+			err := ensureAppNotDeployed(swissKnifeApp)
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
+
+	Context("when I deploy an application", func() {
+		It("should start a deployment", func() {
+			jsonToken := getToken()
+			output, err := cli.Deploy(swissKnifeApp, jsonToken)
+			Expect(err).NotTo(HaveOccurred())
+			expected := "Deployment of application `" + swissKnifeApp.Name + "` has started."
+			Expect(output).To(ContainSubstring(expected))
+			waitTime, _ := time.ParseDuration(InitialWaitTime)
+			time.Sleep(waitTime) // this wait is needed to let the application deploy
+		})
+
+		It("should be in the list of applications in the cluster", func() {
+			list, err := cli.ListApps()
+			Expect(err).NotTo(HaveOccurred())
+			appNames := listAppNames(list)
+			Expect(appNames).To(ContainElement(swissKnifeApp.Name))
+		})
+
+		It("should get to a 'running' status, eventually", func(done Done) {
+			// TODO: check status flag once it's fixed
+			status, err := checkStatusIs(swissKnifeApp, "Running")
+			Expect(err).NotTo(HaveOccurred())
+			Expect(status).To(Equal("Running"))
+			close(done)
+		}, LongTimeout)
+	})
+
+	Context("A deployed test application that uses akka, spark, and flink", func() {
+		It("should contain a spark process", func() {
+			status, err := cli.Status(swissKnifeApp)
+			Expect(err).NotTo(HaveOccurred())
+			streamlets := cli.GetStreamlets(&status)
+			Expect(streamlets).To(ContainElement("spark-process"))
+		})
+
+		It("should contain a flink process", func() {
+			status, err := cli.Status(swissKnifeApp)
+			Expect(err).NotTo(HaveOccurred())
+			streamlets := cli.GetStreamlets(&status)
+			Expect(streamlets).To(ContainElement("flink-process"))
+		})
+
+		It("should contain an akka process", func() {
+			status, err := cli.Status(swissKnifeApp)
+			Expect(err).NotTo(HaveOccurred())
+			streamlets := cli.GetStreamlets(&status)
+			Expect(streamlets).To(ContainElement("akka-process"))
+		})
+
+		It("should produce a counter in the raw output log", func(done Done) {
+			pod, err := cli.GetSinglePodForStreamlet(swissKnifeApp, "raw-egress")
```

Comment:
Does this mean "get one of the (possibly many) pods for this streamlet"? If so, I would rename it to `getOneOfThePodsForStreamlet` or something similar. The word "single" here can read as an expectation that there should only be a single pod, which is probably not true for this app/all apps.

Reply:
done